databroker/tests/test_broker.py

```diff
@@ -509,6 +509,53 @@ class SynWithConfig(Device):
     assert 'exit_status' in ev['timestamps']
 
 
+@py3
+def test_stream_name(db, RE, hw):
+    datum_id = str(uuid.uuid4())
+    datum_id2 = str(uuid.uuid4())
+    desc_uid = str(uuid.uuid4())
+    desc_uid2 = str(uuid.uuid4())
+    event_uid = str(uuid.uuid4())
+    event_uid2 = str(uuid.uuid4())
+
+    # Side-band resource and datum documents.
+    res = db.reg.insert_resource('foo', '', {'x': 1})
+    db.reg.insert_datum(res, datum_id, {'y': 2})
+
+    res2 = db.reg.insert_resource('foo', '', {'x': 1})
+    db.reg.insert_datum(res2, datum_id2, {'y': 2})
+
+    # Generate a normal run.
+    RE.subscribe(db.insert)
+    rs_uid, = RE(count([hw.det]))
+
+    # Side band an extra descriptor and event into this run.
+    data_keys = {'image': {'dtype': 'array', 'source': '', 'shape': [5, 5],
+                           'external': 'FILESTORE://simulated'}}
+    data_keys2 = {'image2': {'dtype': 'array', 'source': '', 'shape': [5, 5],
+                           'external': 'FILESTORE://simulated'}}
+    db.mds.insert_descriptor(run_start=rs_uid, data_keys=data_keys,
+                             time=ttime.time(), uid=desc_uid,
+                             name='injected')
+    db.mds.insert_descriptor(run_start=rs_uid, data_keys=data_keys2,
+                             time=ttime.time(), uid=desc_uid2,
+                             name='injected2')
+    db.mds.insert_event(descriptor=desc_uid, time=ttime.time(), uid=event_uid,
```

Comment:
insert_event is not really necessary. i can remove if needed

Reply:
It is probably simpler to write a custom plan that uses trigger_and_read to add a second stream.

Reply:
@tacaswell This is a different PR. (Maybe I misunderstood)
I was trying to find a way to find fields versus stream_name. I could not find a way without this

Reply:
(this is completely unrelated to streaming for CHX)

Reply:
sorry i understand you now, wrote too quick... will do, thanks!

Reply:
actually, i would argue against this one. This would add in a bluesky dependency. I don't think we should add one if we don't have to. Is this okay as is?

Reply:
We already have a bluesky dependency in this test (the `RE` fixture).

Reply:
Yep, to reiterate some stuff we talked about for the record here...

Originally, all the databroker tests inserted documents manually like this. It was verbose and hard to maintain, and sometimes didn't quite match up with the actual documents generated by bluesky as things changed. Therefore, we decided to use the RunEngine to generate all documents used in databroker tests. We use as a `RE` figure to do that.

A downside of this is that databroker still supports py2, while bluesky only support py3, so tests that use the `RE` fixture cannot be tested on py2. Such tests as decorated with the `@py3` decorator so that the py2 tests skip them. This obviously hurts on py2 test coverage. One possible solution is to generate all the document in Python 3, stash them somewhere online as JSON, and download them for use in the py2 tests. We hope to get around to doing that some day and boosting our py2 test coverage.

Reply:
thanks. i wrote a custom plan that writes to two different streams. tests pass on the test but  `test_find_by_string_time` seems to fail right now. any idea why this could be?